app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,53 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = Recipe.all
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def public
+    @totals = {}
+    @public_recipes = Recipe.where(public: true).order('created_at DESC')
+    @public_recipes.each do |pub|
+      total = 0
+      RecipeFood.where(recipe_id: pub.id).each do |rec_food|
+        total += rec_food.quantity * rec_food.food.price
+      end
+      @totals[pub.name] = total
+    end
+  end
+
+  def create
```

Comment:
There is some issue with creating the recipes. Kindly refer to this [video](https://www.loom.com/share/b395b48615854411aed4705bb64eefdd)for more clarification.